harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -137,19 +143,23 @@ def _init_model(self) -> None:
             self.model = nn.DataParallel(self.model)
             logging.debug("Initialized mode for native parallel training.")
 
-        self.lr_helper = _LRHelper(self.trial.create_lr_scheduler(self.optimizer))
+        self.context.model = self.model
 
-        # If a load path is provided load weights and restore the data location.
-        self._load()
-
-        self._configure_amp()
+    def _init_optimizer(self) -> None:
+        # TODO: Check that optimizer is not an amp optimizer.
+        self.optimizer = self.trial.optimizer(self.model)
```

Comment:
non-blocking: now that we have `self.context` we probably shouldn't have `self.optimizer` and `self.model` anymore.

Reply:
done